pscript/parser1.py

```diff
@@ -408,7 +439,19 @@ def parse_BinOp(self, node):
             return ["Math.pow(", left, ", ", right, ")"]
         elif node.op == node.OPS.FloorDiv:
             return ["Math.floor(", left, "/", right, ")"]
-        
+
+        C = ast.Set
+        if (isinstance(node.left_node, C) and
+                isinstance(node.right_node, C)):
+            if node.op == node.OPS.BitOr:
+                return self.use_std_function('op_set_union', [left, right])
+            if node.op == node.OPS.BitAnd:
+                return self.use_std_function('op_set_intersection', [left, right])
+            if node.op == node.OPS.BitXor:
+                return self.use_std_function('op_set_sym_difference', [left, right])
+            if node.op == node.OPS.Sub:
+                return self.use_std_function('op_set_difference', [left, right])
```

Comment:
I'm a bit unsure about this. It allows set operations on literal sets only. The use is rather limited because in most cases you'd just directly write the result. But more importantly, it may fool users into thinking that operators work for sets, which is not the case (except for literals).

Reply:
I'm happy to drop these for the moment.  They are not in my actual use case.  I just included them to outline how they could be implemented on top of the JS Set.